Refactor docs (#156)
rstudio/rstudio-docker-products: server-pro/README.md

1. These images are provided as a convenience to RStudio customers and are not formally supported by RStudio. If you
   have questions about these images, you can ask them in the issues in the repository or to your support
   representative, who will route them appropriately.
1. Outdated images will be removed periodically from DockerHub as product version updates are made. Please make plans to
   update at times or use your own build of the images.
1. These images are meant as a starting point for your needs. Consider creating a fork of this repo, where you can
   continue to merge in changes we make while having your own security scanning, base OS in use, or other custom
   changes. We
   provide [instructions for building](https://github.com/rstudio/rstudio-docker-products#instructions-for-building) for
   these cases.

#### Simple Example

To verify basic functionality as a first step:

```
# Replace with valid license
export RSP_LICENSE=XXXX-XXXX-XXXX-XXXX-XXXX-XXXX-XXXX

# Run without persistent data using default configuration
docker run --privileged -it \
    -p 8787:8787 \
    -e RSP_LICENSE=$RSP_LICENSE \
    rstudio/rstudio-workbench:latest
```

> Open http://localhost:8787 to access RStudio Server Pro. The default username and password are _rstudio_.

Reviewer: ```suggestion
Open http://localhost:8787 to access RStudio Workbench. The default username and password are _rstudio_.
```

For a more "real" deployment, continue reading!

#### Overview

> Note that running the RStudio Server Pro Docker image requires the container to run using the `--privileged` flag and a

Reviewer: ```suggestion
Note that running the RStudio Workbench Docker image requires the container to run using the `--privileged` flag and a
```

> valid RStudio Server Pro license.

Reviewer: ```suggestion
valid RStudio Workbench license.
```

This container includes:

1. R 3.6.1
2. Python 3.6.5

> 3. RStudio Server Pro

Reviewer: ```suggestion
3. RStudio Workbench
```

You can review possible RStudio Workbench configuration [in the documentation](https://docs.rstudio.com/ide/server-pro/).

See a complete example of server configuration at `server-pro/conf`.

#### Persistent Data

In order to persist user files between container restarts please mount the `/home` directory from a persistent volume on the host
machine or your docker orchestration system.

#### Licensing

The RStudio Workbench Docker image requires a valid license, which can be set in three ways:

1. Setting the `RSP_LICENSE` environment variable to a valid license key inside the container
2. Setting the `RSP_LICENSE_SERVER` environment variable to a valid license server / port inside the container
3. Mounting a `/etc/rstudio-server/license.lic` single file that contains a valid license for RStudio Server Pro

**NOTE:** the "offline activation process" is not supported by this image today. Offline installations will need
to explore using a license server, license file, or custom image with manual intervention.

#### Users

By default, the container will create a test user, which you can control or disable with the environment

> variables: `RSP_TESTUSER`, `RSP_TESTUSER_PASSWD`, `RSP_TESTUSER_UID`.

Reviewer: I assume renaming from RSP will be a part of the workbench conversion?

Author: Yeah, that's a good call... breaking, unfortunately, but we haven't published the workbench images widely yet 🤷‍♂️ (or documented publicly)

For a more "real" deployment, continue reading!

#### Overview

Note that running the RStudio Server Pro Docker image requires the container to run using the `--privileged` flag and a
valid RStudio Server Pro license.

This container includes:

1. R 3.6.1
2. Python 3.6.5
3. RStudio Server Pro

#### Configuration

RStudio Workbench is configured via config files in the in the `/etc/rstudio` directory. Mount this directory as
a volume from the host machine. Changes will take effect when the container is restarted.

Be sure the config files have:

- `launcher.conf` and enable Local Launcher
- `rserver.conf` to connect to the Local Launcher
- `jupyter.conf` with `jupyter-exe` pointing to the Jupyter executable in the Docker image

See a complete example of that file at `server-pro/conf`.

#### Persistent Data

In order to persist user files between container restarts please mount the `/home` directory from a persistent volume on the host
machine or your docker orchestration system.

#### Licensing

The RStudio Workbench Docker image requires a valid license, which can be set in three ways:

> 1. Setting the `RSP_LICENSE` environment variable to a valid license key

Author: ```suggestion
1. Setting the `RSP_LICENSE` environment variable to a valid license key inside the container
```

> 2. Setting the `RSP_LICENSE_SERVER` environment variable to a valid license server / port

Author: ```suggestion
2. Setting the `RSP_LICENSE_SERVER` environment variable to a valid license server / port inside the container
```

> # RStudio Workbench

Reviewer: Should we change the path to workbench?

Reviewer: Replace Server Pro and RSP throughout.

Author: Probably. I was planning to follow up this PR with the change because CI and stuff will need to change too

1. These images are provided as a convenience to RStudio customers and are not formally supported by RStudio. If you
   have questions about these images, you can ask them in the issues in the repository or to your support
   representative, who will route them appropriately.
1. Outdated images will be removed periodically from DockerHub as product version updates are made. Please make plans to
   update at times or use your own build of the images.
1. These images are meant as a starting point for your needs. Consider creating a fork of this repo, where you can
   continue to merge in changes we make while having your own security scanning, base OS in use, or other custom
   changes. We
   provide [instructions for building](https://github.com/rstudio/rstudio-docker-products#instructions-for-building) for
   these cases.

#### Simple Example

To verify basic functionality as a first step:

```
# Replace with valid license
export RSP_LICENSE=XXXX-XXXX-XXXX-XXXX-XXXX-XXXX-XXXX

# Run without persistent data using default configuration
docker run --privileged -it \
    -p 8787:8787 \
    -e RSP_LICENSE=$RSP_LICENSE \
    rstudio/rstudio-workbench:latest
```

> Open http://localhost:8787 to access RStudio Server Pro. The default username and password are rstudio.

Reviewer: ```suggestion
Open http://localhost:8787 to access RStudio Server Pro. The default username and password are _rstudio_.
```

For a more "real" deployment, continue reading!

#### Overview

Note that running the RStudio Server Pro Docker image requires the container to run using the `--privileged` flag and a
valid RStudio Server Pro license.

This container includes:

1. R 3.6.1
2. Python 3.6.5
3. RStudio Server Pro

#### Configuration

> The configuration of RStudio Server Pro is made on a set of file in the `/etc/rstudio` directory. Mount this directory as

Reviewer: ```suggestion
RStudio Workbench is configured via config files in the in the `/etc/rstudio` directory. Mount this directory as
```

> volume with the host machine to change the configuration and restart the container for changes to take effect.

Reviewer: ```suggestion
a volume from the host machine. Changes will take effect when the container is restarted.
```

> Be sure the config files have:

Reviewer: I don't think this belongs here. This is general detail on RSW config, right? There's nothing particualr to the docker config?

Author: Agree - yeah

- `launcher.conf` and enable Local Launcher
- `rserver.conf` to connect to the Local Launcher
- `jupyter.conf` with `jupyter-exe` pointing to the Jupyter executable in the Docker image

See a complete example of that file at `server-pro/conf`.

#### Persistent Data

> In order to persist RSP user files between container restarts please mount `/home` with a persistent volume in the host

Reviewer: ```suggestion
In order to persist user files between container restarts please mount the `/home` directory from a persistent volume on the host
```

machine or your docker orchestration system.

#### Licensing

> Using the RStudio Server Pro Docker image requires to have a valid License. You can set the RSP license to use this in

Reviewer: ```suggestion
The RStudio Workbench Docker image requires a valid license, which can be set in three ways:
```

> one three ways:

Reviewer: ```suggestion
```

1. Setting the `RSP_LICENSE` environment variable to a valid license key
2. Setting the `RSP_LICENSE_SERVER` environment variable to a valid license server / port
3. Mounting a `/etc/rstudio-server/license.lic` single file that contains a valid license for RStudio Server Pro

**NOTE:** the "offline activation process" is not supported by this image today. Offline installations will need
to explore using a license server, license file, or custom image with manual intervention.

#### Users

> By default the container will create a test user, that user can be controlled by the environment

Reviewer: ```suggestion
The container will create a test user when it is created, which you can control with the environment
```

Author: I lean against this because it is also possible to disable this test user

Reviewer: What about if it said, "which you can control or disable with..."

variables: `RSP_TESTUSER`, `RSP_TESTUSER_PASSWD`, `RSP_TESTUSER_UID`.

> In order to use this container with a different user structure such as LDAP you need to extend the container to use a

Reviewer: ```suggestion
This container needs to be extended with a valid PAM configuration if you want to use it with an external user directory such as LDAP/AD.
```

> valid PAM configuration. See

Reviewer: ```suggestion
See
```

the [RStudio Server Pro guide](https://docs.rstudio.com/ide/server-pro/authenticating-users.html) for more information.

#### Environment variables

| Variable | Description | Default |
|-----|---|---|
| `RSP_TESTUSER` | Test user to be created in the container, turn off with an empty value | `rstudio` |
| `RSP_TESTUSER_PASSWD` | Test user password | `rstudio` |
| `RSP_TESTUSER_UID` | Test user UID | `10000` |
| `RSP_LICENSE` | License key for RStudio Server Pro, format should be: `XXXX-XXXX-XXXX-XXXX-XXXX-XXXX-XXXX` | None |
| `RSP_LICENSE_SERVER` | Floating license server, format should be: `my.url.com:port` | None |
| `RSP_LAUNCHER` | Whether or not to use launcher locally / start the launcher process | true |

> | `RSP_LAUNCHER_TIMEOUT` | The timeout, in seconds, to wait for launcher to startup before proceeding | 10 |

Reviewer: This is upon container start? If so...
```suggestion
| `RSP_LAUNCHER_TIMEOUT` | Number of seconds to pause container startup for the launcher process to start | 10 |
```

Reviewer: Not sure I understood quite right what this is...

Author: Yeah, it's not quite "pause container startup." This is basically how long to wait for the launcher to start. Usually it is very quick. If it does not succeed in this amount of time, then the container startup will fail / error.

Author: Tried to add a bit more explanation

Reviewer: Aha -- maybe it'd be clearer to say the timeout is until an error...because it's not a "wait to proceed", it's a "wait to fail", right?

What about "The time, in seconds, the container waits for the launcher process to start before throwing an error"?

Reviewer: "The time, in seconds, to wait for a listening launcher process before throwing an error"?

Reviewer: This really isn't a big deal 😆. I'm sure it's fine as is. I'm just persnickety about writing.
